components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/WebSubHubAdapterUtil.java

```diff
@@ -371,13 +370,18 @@ private static String buildURL(String topic, String webSubHubBaseUrl, String ope
 
     /**
      * Get correlation id from the MDC.
-     * If not then generate a random UUID and return the UUID.
+     * If not then generate a random UUID, add it to MDC and return the UUID.
      *
      * @return Correlation id
      */
     public static String getCorrelationID() {
-
-        return Optional.ofNullable(MDC.get(CORRELATION_ID_MDC)).orElse(UUID.randomUUID().toString());
+        
+        String correlationID = MDC.get(CORRELATION_ID_MDC);
+        if (StringUtils.isBlank(correlationID)) {
+            correlationID = UUID.randomUUID().toString();
+            MDC.put(CORRELATION_ID_MDC, correlationID);
+        }
+        return correlationID;
```

Comment:
why we need this change?

Reply:
If we generate a random UUID as correlation ID and do not add it to the MDC, correlationLog.info() method does not take that ID?